### ISSUE TITLE
CI timeout in test/functional/p2p_feefilter.py

### ISSUE THREAD

**Issue by jonatack** [bitcoin/bitcoin#22483]
https://cirrus-ci.com/task/5047271793360896

```
 test  2021-07-18T01:14:37.374000Z TestFramework.utils (ERROR): wait_until() failed. Predicate: ''''
                                           def test_function():
                                               if check_connected:
                                                   assert self.is_connected
                                               return test_function_in()
                                   '''
 test  2021-07-18T01:14:37.375000Z TestFramework (ERROR): Assertion failed
                                   Traceback (most recent call last):
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 128, in main
                                       self.run_test()
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_feefilter.py", line 64, in run_test
                                       self.test_feefilter()
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_feefilter.py", line 91, in test_feefilter
                                       conn.wait_for_invs_to_match(txids)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/p2p_feefilter.py", line 40, in wait_for_invs_to_match
                                       self.wait_until(lambda: invs_expected == sorted(self.txinvs))
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/p2p.py", line 450, in wait_until
                                       wait_until_helper(test_function, timeout=timeout, lock=p2p_lock, timeout_factor=self.timeout_factor)
                                     File "/tmp/cirrus-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/test/functional/test_framework/util.py", line 261, in wait_until_helper
                                       raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
                                   AssertionError: Predicate ''''
                                           def test_function():
                                               if check_connected:
                                                   assert self.is_connected
                                               return test_function_in()
                                   ''' not true after 2400.0 seconds
```

**Comment by mzumsande** [comment]
I think the reason is that in the failed run the txes rsent to node1 via RPC and relayed to node0 are not accepted to the mempool of node 0 ("bad-txns-premature-spend-of-coinbase") because the sync between the two nodes is not finished by the time they are sent: https://cirrus-ci.com/task/5047271793360896?logs=ci#L2674

While I think that this could be fixed just by adding a `sync_all()` after the `generate` (or by #20362), the root cause for the slow sync seems to be that something with  compact block relay went wrong (many "Large reorg, won't direct fetch" msgs in the log, and it reverts back to inv.) I don't know why that is happening here (unlike in my local runs), the cirrus log unfortunately only has the last 4k rows.

**Comment by jnewbery** [comment]
I think this is fixed by #22567 / #22788.
